### PR TITLE
Account Settings: Add a context menu entry to enable or disable virtual files

### PR DESCRIPTION
**Pull request by ogoffart**
Issue #6725

### CODE REVIEW

**Comment by ogoffart** [comment]
This adds a checkbox in the folder context menu.
The option is only visible in experimental mode.
When enabling it (but not disabling it), it will show the popup which we already show in the wizard.

@michaelstingl 

Screenshot of the menu:

screenshot_20180828_111425

When clicking on the last item, this dialog appear:

screenshot_20180828_111558

File: src/gui/folder.h [hunk 1]
```diff
@@ -244,6 +244,7 @@ class Folder : public QObject
 
     /** new files are downloaded as virtual files */
     bool useVirtualFiles() { return _definition.useVirtualFiles; }
+    void setUseVirtualFiles(bool enabled) { _definition.useVirtualFiles = enabled; }
```
Comment (ckamm) [inline src/gui/folder.h]: Immediate `saveToSettings()` ?

Comment (ogoffart) [reply src/gui/folder.h]: Ooops, i forgot that.

File: src/gui/accountsettings.cpp [hunk 2]
```diff
@@ -315,6 +319,29 @@ void AccountSettings::slotCustomContextMenuRequested(const QPoint &pos)
 
     ac = menu->addAction(tr("Remove folder sync connection"));
     connect(ac, &QAction::triggered, this, &AccountSettings::slotRemoveCurrentFolder);
+
+    if (ConfigFile().showExperimentalOptions() || folder->useVirtualFiles()) {
```
Comment (ckamm) [inline src/gui/accountsettings.cpp]: probably remove `showExperimentalOptions()`

Comment (ckamm) [reply src/gui/accountsettings.cpp]: should become a branding option though

Comment (ckamm) [reply src/gui/accountsettings.cpp]: see #6752

**Comment by ckamm** [comment]
now depends on https://github.com/owncloud/client/pull/6752

**Comment by ckamm** [comment]
@ogoffart merge if it looks good to you

**Comment by ogoffart** [comment]
@michaelstingl  should we merge this into 2.5.0?
Experimental settings always visible in the wizard is one thing, but always visible in the context menu is a different thing.

**Comment by michaelstingl** [comment]
@ogoffart I also don't feel confident having this in 2.5.0.

**Comment by ogoffart** [comment]
@michaelstingl  Do we want this patch in 2.5.1 ?  Or should we wait even more?

**Comment by guruz** [comment]
I think we should have this in 2.5.1.
@ogoffart Do you need to re-base since https://github.com/owncloud/client/pull/6752 is merged? (github claims it can still merge)

**Comment by michaelstingl** [comment]
okay, let's try this 👍

**Pull request merged**